[🐛 Bug]: desync puppeteer-core peer-dep version

### Have you read the Contributing Guidelines on issues?

- [x] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

latest

### Mode

WDIO Testrunner

### What happened?

`webdriverio` package is using:

- `"puppeteer-core": "^23.0.2"` as dev dependency
https://github.com/webdriverio/webdriverio/blob/24fb81f01c2ca1ec6a26dc04b9c7feca6c2b6f08/packages/webdriverio/package.json#L74

but also specify:

- `"puppeteer-core": "^22.3.0"` as peer dependency
https://github.com/webdriverio/webdriverio/blob/24fb81f01c2ca1ec6a26dc04b9c7feca6c2b6f08/packages/webdriverio/package.json#L105

Which makes parent project like mine forced to sit on v22.

Is there any reason not to align the peer dep with the dev dep ?

Thanks

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

npm prints warnings when trying to install 23 version of Puppeteer:

```bash
node -v && npm -v
```

```bash
v22.14.0
10.9.2
```

```bash
mkdir wdio-p
cd wdio-p

npm init wdio@latest . -- --yes
npm i puppeteer-core@23
```

```
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: webdriverio@9.12.4
npm warn Found: puppeteer-core@23.11.1
npm warn node_modules/puppeteer-core
npm warn   puppeteer-core@"23" from the root project
npm warn
npm warn Could not resolve dependency:
npm warn peerOptional puppeteer-core@"^22.3.0" from webdriverio@9.12.4
npm warn node_modules/webdriverio
npm warn   webdriverio@"9.12.4" from @wdio/cli@9.12.4
npm warn   node_modules/@wdio/cli
npm warn   3 more (@wdio/globals, @wdio/runner, expect-webdriverio)
npm warn
npm warn Conflicting peer dependency: puppeteer-core@22.15.0
npm warn node_modules/puppeteer-core
npm warn   peerOptional puppeteer-core@"^22.3.0" from webdriverio@9.12.4
npm warn   node_modules/webdriverio
npm warn     webdriverio@"9.12.4" from @wdio/cli@9.12.4
npm warn     node_modules/@wdio/cli
npm warn     3 more (@wdio/globals, @wdio/runner, expect-webdriverio)
```

--- a/packages/webdriverio/package.json
+++ b/packages/webdriverio/package.json
@@ -102,7 +102,7 @@
     "webdriver": "workspace:*"
   },
   "peerDependencies": {
-    "puppeteer-core": "^22.3.0"
+    "puppeteer-core": ">=22.x || <=24.x"
   },
   "peerDependenciesMeta": {
     "puppeteer-core": {

